src/Aks/Aks/Commands/CreateOrUpdateKubeBase.cs

```diff
@@ -352,7 +352,8 @@ private AcsServicePrincipal BuildServicePrincipal(string name, string clientSecr
                     desensitizedMessage: Resources.CouldNotCreateAServicePrincipalWithTheRightPermissionsAreYouAnOwner);
             }
 
-            AddSubscriptionRoleAssignment("Contributor", sp.Id);
+            // Adding Contributor access to the ManagedCluster resourceGroup
+            AddClusterResourceGroupRoleAssignment("Contributor", sp.Id, this.ResourceGroupName);
```

Comment:
These changes in this file should be in another PR and we will merge it in the next major release branch.